### DISCUSSION TITLE
Failed to create session.

### DISCUSSION THREAD

**Discussion by miron50** [appium/appium#18930] [Q&A] [answered]
Failed to create session. An unknown server-side error occurred while processing the command. Original error: Cannot verify the signature of 'D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-uiautomator2-server\apks\appium-uiautomator2-server-v4.27.0.apk'. Original error: java.lang.UnsupportedClassVersionError: com/android/apksigner/ApkSignerTool : Unsupported major.minor version 52.0 at java.lang.ClassLoader.defineClass1(Native Method) at java.lang.ClassLoader.defineClass(ClassLoader.java:800) at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142) at java.net.URLClassLoader.defineClass(URLClassLoader.java:449) at java.net.URLClassLoader.access$100(URLClassLoader.java:71) at java.net.URLClassLoader$1.run(URLClassLoader.java:361) at java.net.URLClassLoader$1.run(URLClassLoader.java:355) at java.security.AccessController.doPrivileged(Native Method) at java.net.URLClassLoader.findClass(URLClassLoader.java:354) at java.lang.ClassLoader.loadClass(ClassLoader.java:425) at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) at java.lang.ClassLoader.loadClass(ClassLoader.java:358) at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482) Exception in thread "main"

How to solve this error?

**Appium server log**

[BaseDriver] Session created with session id: 5cdc7c8c-8b32-4ee0-90ce-8635ba6ba574
[UiAutomator2] Starting 'com.google.android.Calculator' directly on the device
[ADB] Using 'adb.exe' from 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe'
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 start-server'
[AndroidDriver] Retrieving device list
[ADB] Trying to find a connected android device
[ADB] Getting connected devices
[ADB] Connected devices: [{"udid":"320204445594","state":"device"}]
[AndroidDriver] Using device: 320204445594
[ADB] Using 'adb.exe' from 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe'
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 start-server'
[ADB] Setting device id to 320204445594
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell getprop ro.build.version.sdk'
[ADB] Current device property 'ro.build.version.sdk': 28
[ADB] Getting device platform version
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell getprop ro.build.version.release'
[ADB] Current device property 'ro.build.version.release': 9
[ADB] Device API level: 28
[UiAutomator2] Relaxing hidden api policy
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell 'settings put global hidden_api_policy_pre_p_apps 1;settings put global hidden_api_policy_p_apps 1;settings put global hidden_api_policy 1''
[AndroidDriver] No app sent in, not parsing package/activity
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 wait-for-device'
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell echo ping'
[AndroidDriver] Pushing settings apk to device...
[ADB] Getting install status for io.appium.settings
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell dumpsys package io.appium.settings'
[ADB] 'io.appium.settings' is installed
[ADB] Getting package info for 'io.appium.settings'
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell dumpsys package io.appium.settings'
[ADB] The version name of the installed 'io.appium.settings' is greater or equal to the application version name ('3.4.0' >= '3.4.0')
[ADB] There is no need to install/upgrade 'D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\io.appium.settings\apks\settings_apk-debug.apk'
[ADB] Getting IDs of all 'io.appium.settings' processes
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell 'pgrep --help; echo $?''
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[AndroidDriver] io.appium.settings is already running. There is no need to reset its permissions.
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell appops set io.appium.settings android:mock_location allow'
[Logcat] Starting logs capture with command: 'C:\\Program Files\\Android\\Android Studio\\platform-tools\\adb.exe' -P 5037 -s 320204445594 logcat -v threadtime
[UiAutomator2] Forwarding UiAutomator2 Server port 6790 to local port 8200
[ADB] Forwarding system: 8200 to device: 6790
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 forward tcp:8200 tcp:6790'
[ADB] Getting install status for io.appium.uiautomator2.server
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell dumpsys package io.appium.uiautomator2.server'
[ADB] 'io.appium.uiautomator2.server' is not installed
[ADB] App 'D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-uiautomator2-server\apks\appium-uiautomator2-server-v4.27.0.apk' is not installed
[UiAutomator2] io.appium.uiautomator2.server installation state: notInstalled
[ADB] Checking app cert for D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-uiautomator2-server\apks\appium-uiautomator2-server-v4.27.0.apk
[ADB] Using 'apksigner.jar' from 'C:\Program Files\Android\Android Studio\build-tools\34.0.0\lib\apksigner.jar'
[ADB] Starting apksigner: 'C:\\Program Files\\Java\\jdk1.7.0_80\\bin\\java.exe' -Xmx1024M -Xss1m -jar 'C:\\Program Files\\Android\\Android Studio\\build-tools\\34.0.0\\lib\\apksigner.jar' verify --print-certs D:\\Install\\Appium-Server-GUI-windows-1.22.3-4\\resources\\app\\node_modules\\appium\\node_modules\\appium-uiautomator2-server\\apks\\appium-uiautomator2-server-v4.27.0.apk
[UiAutomator2] Deleting UiAutomator2 session
[UiAutomator2] Deleting UiAutomator2 server session
[WD Proxy] Matched '/' to command name 'deleteSession'
[UiAutomator2] Did not get confirmation UiAutomator2 deleteSession worked; Error was: UnknownError: An unknown server-side error occurred while processing the command. Original error: Trying to proxy a session command without session id
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell am force-stop com.google.android.Calculator'
[Logcat] Stopping logcat capture
[ADB] Removing forwarded port socket connection: 8200 
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 forward --remove tcp:8200'
[UiAutomator2] Restoring hidden api policy to the device default configuration
[ADB] Running 'C:\Program Files\Android\Android Studio\platform-tools\adb.exe -P 5037 -s 320204445594 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy''
[BaseDriver] Event 'newSessionStarted' logged at 1690816266223 (17:11:06 GMT+0200 (Central European Summer Time))
[W3C] Encountered internal error running command: Error: Cannot verify the signature of 'D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-uiautomator2-server\apks\appium-uiautomator2-server-v4.27.0.apk'. Original error: java.lang.UnsupportedClassVersionError: com/android/apksigner/ApkSignerTool : Unsupported major.minor version 52.0
[W3C] 	at java.lang.ClassLoader.defineClass1(Native Method)
[W3C] 	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
[W3C] 	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
[W3C] 	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
[W3C] 	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
[W3C] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
[W3C] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
[W3C] 	at java.security.AccessController.doPrivileged(Native Method)
[W3C] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
[W3C] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
[W3C] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
[W3C] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
[W3C] 	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)
[W3C] Exception in thread "main" 
[W3C]     at ADB.checkApkCert (D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-adb\lib\tools\apk-signing.js:306:11)
[W3C]     at UiAutomator2Server.installServerApk (D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-uiautomator2-driver\lib\uiautomator2.js:117:13)
[W3C]     at AndroidUiautomator2Driver.initUiAutomator2Server (D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-uiautomator2-driver\lib\driver.js:494:7)
[W3C]     at AndroidUiautomator2Driver.startUiAutomator2Session (D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-uiautomator2-driver\lib\driver.js:390:5)
[W3C]     at AndroidUiautomator2Driver.createSession (D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\node_modules\appium-uiautomator2-driver\lib\driver.js:229:7)
[W3C]     at AppiumDriver.createSession (D:\Install\Appium-Server-GUI-windows-1.22.3-4\resources\app\node_modules\appium\lib\appium.js:387:35)
[HTTP] <-- POST /wd/hub/session 500 3919 ms - 3091
[HTTP]

**Comment by jlipps** [comment, marked as answer]
I googled the error message and got links like this: https://rollbar.com/blog/java-unsupported-major-minor-version-52-0/. Is your version of Java too old?

**Comment by miron50** [comment]
I installed jdk-20, update JAVA_HOME, PATH, but issue is reproduced

**Comment by miron50** [comment]
Again install UiAutomarot2, change the launching application to Chrome, and try to use the virtual device instead of the real device and magic.
